tests/test_slips_utils.py

```diff
@@ -1,10 +1,410 @@
+
+
+def test_get_first_octet():
+    utils = ModuleFactory().create_utils_obj()
+
+    # IPv4 address
+    assert utils.get_first_octet("192.168.1.100") == "192"
+    # IPv6 address
+    assert (
+        utils.get_first_octet("2001:0db8:85a3:0000:0000:8a2e:0370:7334")
+        == "2001"
+    )
+    # invalid IP address
+    assert utils.get_first_octet("invalid") is None
+
+
+@pytest.mark.parametrize(
+    "input_value, expected_output",
+    [
+        (0, 0.3),
+        (5, 0.5),
+        (10, 1.0),
+        (15, 1.0),
+        (-5, -0.5),
+        (1000000, 1),
+        (sys.maxsize, 1.0),
+    ],
+)
+def test_calculate_confidence(input_value, expected_output):
+    utils = ModuleFactory().create_utils_obj()
+    assert utils.calculate_confidence(input_value) == expected_output
+
+    if input_value < 0:
+        assert (
+            utils.calculate_confidence(input_value) == -0.5
+        ), "Confidence for negative packet counts should be 0."
+    elif input_value > 1000000:
+        assert (
+            utils.calculate_confidence(input_value) == 1
+        ), "Confidence for large packet counts should be capped at 1."
+    elif input_value == sys.maxsize:
+        assert (
+            utils.calculate_confidence(input_value) == 1.0
+        ), "Confidence should be capped at 1.0 for extremely large packet counts"
+
+
+def test_convert_format():
+    utils = ModuleFactory().create_utils_obj()
+    assert (
+        utils.convert_format(
+            "2023-04-06T12:34:56.789Z", "%Y-%m-%dT%H:%M:%S.%fZ"
+        )
+        == "2023-04-06T12:34:56.789000Z"
+    )
+    assert (
+        utils.convert_format(1680788096.789, "iso")
+        == "2023-04-06T13:34:56.789000+00:00"
+    )
+    assert (
+        utils.convert_format(1680788096.789, "%Y-%m-%d %H:%M:%S")
+        == "2023-04-06 19:04:56"
+    )
+    assert (
+        utils.convert_format(
+            datetime.datetime(2023, 4, 6, 12, 34, 56, 789000), "unixtimestamp"
+        )
+        == 1680764696.789
+    )
+
+
+def test_assert_microseconds():
+    utils = ModuleFactory().create_utils_obj()
+    assert utils.assert_microseconds("1680788096.789") == "1680788096.789000"
+    assert utils.assert_microseconds("1680788096") == "1680788096"
+    assert (
+        utils.assert_microseconds("1680788096.123456") == "1680788096.123456"
+    )
+    assert (
+        utils.assert_microseconds("1680788096.123456789")
+        == "1680788096.123456789"
+    )
+
+
+@pytest.mark.parametrize(
+    "input_value, expected_output",
+    [
+        (0.1, "low"),
+        (0.4, "medium"),
+        (0.5, "medium"),
+        (0.7, "high"),
+        (1.0, "critical"),
+        (-1, "info"),
+        (2, None),
+    ],
+)
+def test_threat_level_to_string(input_value, expected_output):
+    # utils = ModuleFactory().create_utils_obj()
+    ...
+
+
+def test_is_private_ip():
+    utils = ModuleFactory().create_utils_obj()
+    assert utils.is_private_ip(ipaddress.ip_address("192.168.1.1"))
```

Comment:
we can mark.parametrize here. we can use it anywhere where there's repetition. you can check the rest of the file, you get the point :'D

Reply:
Okay I will use parametrize in this cases and update this PR

Reply:
not resolved yet right?